src/OpenTelemetry.ResourceDetectors.Azure/AzureVmMetaDataRequestor.cs

```diff
@@ -28,7 +28,7 @@ internal static class AzureVmMetaDataRequestor
 
     public static AzureVmMetadataResponse? GetAzureVmMetaDataResponseDefault()
     {
-        using var httpClient = new HttpClient();
+        using var httpClient = new HttpClient() { Timeout = TimeSpan.FromSeconds(2) };
```

Comment:
We might need to consider a 1 sec, it's good to start with 2 secs delay.

Reply:
Why are we changing the default value?

Reply:
The default timeout 100 secs. For applications running in non-Azure VM environments this results in longer start up time for app.

Reply:
Would this throw an exception when timed out?

Reply:
Yes.

Reply:
Okay just checked that it would be caught in `Detect`.